msm/src/circuit_design/witness.rs

```diff
@@ -67,7 +70,11 @@ impl<
     type Variable = F;
 
     fn assert_zero(&mut self, cst: Self::Variable) {
-        assert_eq!(cst, F::zero());
+        assert_eq!((self.assert_mapper)(cst), F::zero());
```

Comment:
Ah, is `assert_mapper` only used for the `assert_zero`, so you can constrain multiple expressions at the same time for a given selector, like in constraint_ivc?